tools/docker/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM ubuntu
+
+ARG USER_ID
+ARG GROUP_ID
+
+RUN apt-get update
+RUN apt-get install -y python3-pep517 python3-can python3-serial python3-setuptools python3-pyelftools python3-qtpy python3-wheel libtinfo5
+RUN apt-get install -y curl patch git
+
+# RUN addgroup --gid $GROUP_ID user
```

Comment:
I believe this is required for Linux to avoid leaving root turds around the place.
I'm not entirely sure off the bat how best do deal with the differences in OSx vs. Linux on this one.

Additionally, I think there's some potential Windows complications too, but I'm less familiar with them.